apps/omg_watcher_rpc/lib/web/validators/order.ex

```diff
@@ -45,6 +46,19 @@ defmodule OMG.WatcherRPC.Web.Validator.Order do
     end
   end
 
+  defp ensure_not_self_transaction(owner, payments) when length(payments) > 0 do
+    payments
+    |> Enum.any?(fn payment ->
+      owner != payment[:owner]
+    end)
+    |> handle_self_tx_result()
+  end
+
+  defp ensure_not_self_transaction(_, _), do: :ok
```

Comment:
curious, when/how would this get hit?

Reply:
So it's actually possible to have an empty list of payments at that point, as demonstrated by the two tests below:

- https://github.com/omisego/elixir-omg/blob/534ed37f61b4ad4722284c01f7343e0d8a5e579b/apps/omg_watcher_rpc/test/omg_watcher_rpc/web/controllers/transaction_test.exs#L1171-L1201
- https://github.com/omisego/elixir-omg/blob/534ed37f61b4ad4722284c01f7343e0d8a5e579b/apps/omg_watcher_rpc/test/omg_watcher_rpc/web/controllers/transaction_test.exs#L1204-L1231

I'm not sure why we don't validate it earlier (and I'm kinda afraid of changing it), but it gets checked later in the UTXO selection, so I had to handle that with this clause. :/

Reply:
hmmm `empty transaction without payments list is not allowed` seems like if you had a clause `ensure_not_empty`, this would cover it before it hits OMG.State, right?

Reply:
interesting. So the idea is to allow it to bypass the check here and let it fail later. I see.

Reply:
@InoMurko yep, that's right. I don't see a reason why we couldn't validate it early in `order.ex`, but I'm afraid I'm lacking some context. Maybe we could create an issue about it and get it fixed in another PR?

Reply:
Sure! Can you open it?

Reply:
Issue created: https://github.com/omisego/elixir-omg/issues/1419